zebra/kernel_netlink.c

```diff
@@ -394,6 +394,15 @@ static int rt_netlink_socket(struct nlsock *nl, unsigned long groups,
 			      NETLINK_ROUTE);
 }
 
+/* Make socket for Linux generic netlink interface. */
+static int ge_netlink_socket(struct nlsock *nl, unsigned long groups,
```

Comment:
again too much abstraction.  Just call the netlink_socket with the NETLINK_GENERIC